src/lwc/geFormWidgetTokenizeCard/geFormWidgetTokenizeCard.js

```diff
@@ -33,6 +36,67 @@ export default class geFormWidgetTokenizeCard extends LightningElement {
     CUSTOM_LABELS = GeLabelService.CUSTOM_LABELS;
     PAYMENT_TRANSACTION_STATUS_ENUM;
 
+    _currentPaymentMethod = undefined;
+    _hasPaymentMethodInTemplate = false;
+
+    @api
+    get widgetDataFromState() {
+        return this._widgetDataFromState;
+    }
+
+    set widgetDataFromState(widgetState) {
+        this._widgetDataFromState = widgetState;
+        this.handleWidgetDataChange(widgetState);
+    }
+
+    handleWidgetDataChange(widgetState) {
+        this._hasPaymentMethodInTemplate =
+            this.sourceFieldsUsedInTemplate.includes(apiNameFor(DATA_IMPORT_PAYMENT_METHOD));
+
+        if (this._hasPaymentMethodInTemplate) {
+            this._currentPaymentMethod = widgetState[apiNameFor(DATA_IMPORT_PAYMENT_METHOD)];
+
+            if (this.hasValidPaymentMethod(this._currentPaymentMethod)) {
+                if (this.isMounted) {
+                    this.requestSetPaymentMethod(this._currentPaymentMethod);
+                } else {
+                    this.handleUserEnabledWidget();
+                }
+            } else {
+                this.handleUserDisabledWidget();
+            }
+        }
+    }
+
+    hasValidPaymentMethod(paymentMethod) {
+        return paymentMethod === 'ACH' || paymentMethod === 'Credit Card';
+    }
+
+    requestSetPaymentMethod(paymentMethod) {
+        this.isLoading = true;
+        const iframe = this.template.querySelector(`[data-id='${this.CUSTOM_LABELS.commonPaymentServices}']`);
+        tokenHandler.setPaymentMethod(iframe, paymentMethod, this.handleError, this.resolveSetPaymentMethod);
+    }
+
+    resolveSetPaymentMethod = () => {
+        this.isLoading = false;
+    }
+
+    get shouldDisplayEnableButton() {
+        if (!this._hasPaymentMethodInTemplate) return true;
```

Comment:
Can we combine these two `if` statements, maybe something like this: 
```
  return !!(!this._hasPaymentMethodInTemplate ||
            (this._hasPaymentMethodInTemplate &&
                this.hasValidPaymentMethod(this._currentPaymentMethod)));
    
```

Reply:
To me, it's just a little less readable at a glance but if another dev agrees that we should make this change then I'd be happy to do it!

Reply:
The guidance is to store Boolean expressions into vars so they are more readable. Putting everything into one statement does not seem readable to me, and I were debugging this, it would take me a while to understand the expression, so I would suggest not to go down that route.